crates/ty_ide/src/inlay_hints.rs

```diff
@@ -86,6 +87,15 @@ pub struct InlayHintSettings {
     pub function_argument_names: bool,
 }
 
+impl Default for InlayHintSettings {
+    fn default() -> Self {
+        Self {
+            variable_types: true,
+            function_argument_names: true,
+        }
+    }
+}
```

Comment:
I'm not entirely sure about enabling all of them by default. E.g. pylance all inlays by default (which I'm also not sure if that's great). We probably want something in between where we enable the non noisy inlays by default but keep all others disabled. 

I assume the new default implementation matches the defaults in the VS code settings?

Reply:
As discussed in our 1:1, we've decided to keep it enabled by default, and can change it in the future before GA based on user feedback.

Reply:
Oh, I think I forgot about that discussion, thanks for the reminder. I'll revert this change back for now.

Reply:
> Does this behaviour match Pylance? 

It does not. Pylance has them off by default. 

I still think we should keep inlays that we find useful and aren't noisy on by default. That's what users opt-in to by enabling inlay hints at the editor level and I rather change our defaults based on user feedback. Which I think we get more of if we enable them by default and it's something we can change very easily moving forward

Reply:
Just for context, the variable type hints were enabled by default when it got added and it only got disabled by default in https://github.com/astral-sh/ruff/pull/19780 which I did not intend to make. That was the main motivation of this PR.